src/app/plugins/marketplace.plugin.ts

```diff
@@ -93,14 +94,30 @@ export class MarketPlacePlugin extends Plugin {
         if (!startsWithPrefix) {
           return;
         }
+        last_id = msg.id; //last_id will end up as the last message's id
 
         let [, item] = content.split('add');
         if (item?.length) {
           const user = msg.author;
           item += '\n' + user; //adds user to end of listing.
-          itemsForSale.push(item);
+
+          //Check if sold
+          const hasTargetReaction = msg.reactions.find(
+            (r) => r.emoji.name === this._TARGET_REACTION
+          );
+
+          if (hasTargetReaction === null) {
+            itemsForSale.push(item);
+            return;
+          }
+
+          hasTargetReaction.fetchUsers().then((users) => {
+            if (users.has(msg.author.id)) {
+              item += '\t SOLD';
+            }
+            itemsForSale.push(item);
+          });
```

Comment:
do we ever await this?

Reply:
You mean i shouldn't await it?

Reply:
you have to await it because you need the push to happen before you build the message right?

Reply:
Correct. I used then, doesn't that work?

Reply:
`then` unwraps the promise, does some stuff, and then returns a new promise. So it doesn't necessarily work afaik.

Reply:
Oh, ok. I was having trouble using await inside of a forEach, I'll make a queue or something instead